### PR TITLE
Ethereum transactions

### PR DESCRIPTION
**Pull request by eldargab**

### CODE REVIEW

**Comment by eldargab** [comment]
Hey, @belopash , @tmcgroul !

Shall we support `sighash` and wildcard contract filters right away?

**Comment by belopash** [comment]
It would be nice

> Hey, @belopash , @tmcgroul !
> 
> Shall we support `sighash` and wildcard contract filters right away?

File: substrate-processor/src/interfaces/gateway.ts [hunk 1]
```diff
@@ -9,7 +9,7 @@ export interface BatchRequest {
     events?: ObjectRequest[]
     calls?: ObjectRequest[]
     evmLogs?: EvmLogRequest[]
-    ethereumTransactions?: EthereumTransactionRequest[]
+    ethereumTransactCalls?: EthereumTransactRequest[]
```
Comment (belopash) [inline substrate-processor/src/interfaces/gateway.ts]: I think we should stick to one name `ethereumTransactions` or `ethereumTransactCalls`, overwise there can be confusion due different names in different parts of code

Comment (belopash) [reply substrate-processor/src/interfaces/gateway.ts]: oh, was fixed in commite above

**Comment by tmcgroul** [comment]
> Hey, @belopash , @tmcgroul !
> 
> Shall we support `sighash` and wildcard contract filters right away?
What if we add `sighash` support in a future release? Will we have to resync archive?

**Comment by eldargab** [comment]
No, `sighash` is already archived by this PR.

**Comment by tmcgroul** [comment]
In that case let's support `sighash`. I'll update the gateway

**Comment by tmcgroul** [comment]
how about such interface:
`ethereumTransactions: [{contract: "*", selector: "0x095ea7b3"}]`?

**Comment by eldargab** [comment]
I'm already putting `sighash` everywhere. Any reason why `selector` is better?

**Comment by tmcgroul** [comment]
Only because i've already put `selector` in acala handlers.
Ok let it be sighash

**Comment by belopash** [comment]
Has moonriver archive been migrated to new release? If yes, I think it's better to merge two eth test projects

**Pull request merged**